corehq/apps/locations/permissions.py

```diff
@@ -20,34 +21,57 @@ def is_locations_admin(view_fn):
     return locations_access_required(domain_admin_required(view_fn))
 
 
-def user_can_edit_location(user, location, project):
-    """
-    Expects SQLLocation
-    """
-    if not project.location_restriction_for_users:
+@quickcache(['user._id', 'project.name'])
```

Comment:
What was the reasoning for adding this? It causes a 5 minute delay on new locations showing up in the location list UI, which is being reported as a bug.

@emord @czue